spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,136 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe PromptsController do
   include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+  let(:open_signup) do
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
 
   before do
     fake_login
   end
 
-  describe 'no_prompt' do
-    it 'should show an error and redirect' do
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to collection_path(collection)
+      expect(flash[:error]).to eq "What challenge did you want to sign up for?"
```

Comment:
With RedirectExpectationHelper, it should be possible to write one line to check anything that redirects with an error or notice. For an error, it would be: `it_redirects_to_with_error(collection_path(collection), "What challenge did you want to sign up for?")`